app/controllers/users.js

```diff
@@ -87,9 +89,21 @@ const findAll = (request, response) =>
 
 const findUserAlbums = (request, response) =>
   users
-    .findById(request.params.user_id)
+    .findById(request.params.id)
     .then(user => user.getAlbums())
     .then(albums => response.send(albums))
     .catch(_logDatabaseError(response));
 
-module.exports = { save, login, findAll, saveOrUpdateAdmin, findUserAlbums };
+const findUserAlbumsPhotos = (request, response) =>
+  users
+    .findById(request.params.id)
+    .then(user => user.getAlbums())
+    .then(albums =>
+      Promise.all(
+        Array.prototype.map.call(albums, album => axios.get(`${photoEndpoint}?albumId=${album.id}`))
+      )
+    )
+    .then(photosMatrix => response.send([].concat(...photosMatrix.map(res => res.data))))
```

Comment:
you dont need [].concat, do you?

Reply:
Hm, I think I do. In this case "photoMatrix" it's and array of arrays because the response for `photoEndpoint` is an array, so without the `concat`, it'll call `.send()` with multiple params and it'll consider only the first array as body for the response.

![image](https://user-images.githubusercontent.com/39705463/41438838-af056a86-6ff6-11e8-9ca1-43ebd80d9dbc.png)

Reply:
and what about a flatmap?

Reply:
It didn't work. It seems that is not supported in nodeJS (sadly... It looks great)
![image](https://user-images.githubusercontent.com/39705463/41440606-2839683e-6ffd-11e8-8d2e-86bbbc219d80.png)

`map().flat()` dind't work either :(

Reply:
its ok for now, we use `lodash` for this cases. It has a lot of useful functions, like flatMap :) . I dont know why it's not supported in node.